source/localizable/tutorial/autocomplete-component.md

```diff
@@ -193,11 +200,78 @@ export default function() {
 }
 ```
 
-After updating our mirage configuration, we should see passing tests, as well as a simple filter on your home screen,
-that will update the rental list as you type:
+After updating our mirage configuration, we should see a simple filter on the home screen that will update the rental list as you type:
 
 ![home screen with filter component](../../images/autocomplete-component/styled-super-rentals-filter.png)
 
+#### Handling Results Coming Back at Different Times
+
+In our example, you might notice that if you type quickly that the results might get out of sync with the current filter text entered.
+This is because our data filtering function is _asynchronous_, meaning that the code in the function gets scheduled for later, while the code that calls the function continues to execute.
+Often code that may make network requests is set up to be asynchronous because the server may return its responses at varying times.
+
+Lets add some protective code to ensure our results do not get out of sync with our filter input.
+To do this we'll simply provide the filter text to the filter function, so that when the results come back we can compare the original filter value with the current filter value.
+We will update the results on screen only if the original filter value and the current filter value are the same.
+
+```app/controllers/rentals.js{+7,+8,+9,+10,+12,+13,+14,+15}
+import Ember from 'ember';
+
+export default Ember.Controller.extend({
+  actions: {
+    filterByCity(param) {
+      if (param !== '') {
+        return this.get('store')
+          .query('rental', { city: param }).then((results) => {
+            return { query: param, results: results };
+          });
+      } else {
+        return this.get('store')
+          .findAll('rental').then((results) => {
+            return { query: param, results: results };
+          });
+      }
+    }
+  }
+});
+```
+
+In the `filterByCity` function in the rental controller above,
+we've added a new property called `query` to the filter results instead of just returning an array of rentals as before.
+
+```app/components/list-filter.js{+9,+10,+11,+19,+20,+21}
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  classNames: ['list-filter'],
+  value: '',
+
+  init() {
+    this._super(...arguments);
+    this.get('filter')('').then((allResults) => {
```

Comment:
This is more for another pull request, but might be useful if we set the search value on the query params and then use it here to retrieve the initial search results